packages/amplication-server/src/core/auth/auth.service.ts

```diff
@@ -423,6 +527,72 @@ export class AuthService {
     return workspace as unknown as Workspace & { users: AuthUser[] };
   }
 
+  private async createPreviewWorkspaceWithProjectAndResource(account: Account) {
+    const workspaceName = `Amplication-${this.generateRandomString()}`;
+    const workspace = (await this.workspaceService.createPreviewWorkspace(
+      {
+        data: {
+          name: workspaceName,
+        },
+        include: WORKSPACE_INCLUDE,
+      },
+      account.id
+    )) as unknown as Workspace & { users: AuthUser[] };
+    const [user] = workspace.users as AuthUser[];
+
+    const project = await this.projectService.createPreviewProject(
+      {
+        data: {
+          name: account.previewAccountType,
+          workspace: {
+            connect: {
+              id: workspace.id,
+            },
+          },
+        },
+      },
+      user.id
+    );
+
+    const resource = await this.resourceService.createPreviewService({
+      args: {
+        data: {
+          name: "Monolith",
+          description: "Monolith Service",
+          resourceType: EnumResourceType.Service,
+          project: {
+            connect: {
+              id: project.id,
+            },
+          },
+          serviceSettings: {
+            authProvider: EnumAuthProviderType.Jwt,
+            authEntityName: "User",
+            adminUISettings: {
+              adminUIPath: "admin",
+              generateAdminUI: true,
+            },
+            serverSettings: {
+              generateGraphQL: true,
+              generateRestApi: true,
+              serverPath: "server",
+            },
+          },
+        },
+      },
+      user,
+      pluginsToInstall: [],
+      requireAuthenticationEntity: true,
```

Comment:
What does it mean not to install any plugin but mandate "requireAuthenticationEntity"? Would it create the user entity? Would it force the user to create one? 
Don't we want to install the default auth plugins?

Reply:
plugins to install are "other plugins". This is for future usage. Have a look at the `createPrevewService` implementation. We can move the logic to here but I though that here we just need to send a list of plugins to install that are not the "default". The resource service handles the default behavior

Reply:
I wrote a separate comment on that, so I am moving the discussion there and marking this one as resolved. However, I think if the parameter name would change from `pluginsToInstall` to something like `nonDefaultPluginsToInstal` or `additionalPluginsToInstall` it will become more readable